OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java

```diff
@@ -339,6 +346,60 @@ public void testAmazonDoesNotGetAdId() throws Exception {
       assertFalse(ShadowAdvertisingIdProviderGPS.calledGetIdentifier);
    }
 
+   @Test
+   public void testGenerateJsonDiffMutate() throws Exception {
```

Comment:
since it's a new block of unit test, and to continue making MainOneSignalClassRunner smaller, can you remove this UnitTest to a new file like UserStateTests.java?